### PR TITLE
spammer: support for standalone benchmarking

### LINKED ISSUE #1370: Make bench tool able to run standalone as well

### PR DESCRIPTION
**Pull request by anwayde**
Closes #1370

### CODE REVIEW

**Comment by mmcgee-jump** [comment]
I think my main hesitation is growing the config TOML for new commands. It's more for configuring the production validator, and I stuck the bench stuff in there as a bit of a convenience because it's reasonably related when you are benching the validator that you plan to run.

For this command, though, it stands on its own. I think we could rename it `spammer` maybe, and make the arguments command line arguments, rather than being in the TOML file.

File: src/disco/topo/fd_topob.c [hunk 1]
```diff
@@ -120,7 +120,7 @@ fd_topob_tile( fd_topo_t *    topo,
   if( FD_UNLIKELY( topo->tile_cnt>=FD_TOPO_MAX_TILES ) ) FD_LOG_ERR(( "too many tiles" ));
 
   ulong kind_id = 0UL;
-  for( ulong i=0UL; i<topo->link_cnt; i++ ) {
```
Comment (mmcgee-jump) [inline src/disco/topo/fd_topob.c:123]: Can you make this a separate PR? Think it should go in ASAP.

File: src/app/fddev/tiles/fd_benchs.c [hunk 2]
```diff
@@ -86,7 +86,7 @@ privileged_init( fd_topo_t *      topo,
 
   ushort port = 12000;
 
-  ctx->conn_cnt = fd_topo_tile_name_cnt( topo, "quic" );
```
Comment (mmcgee-jump) [inline src/app/fddev/tiles/fd_benchs.c:89]: This change makes sense for the spammer, but not when running the existing bench command, since it should auto detect this from the topo still.

**Review by mmcgee-jump — CHANGES_REQUESTED** [review]

File: src/app/fddev/tiles/fd_benchs.c (hunk 2 above)
Comment (anwayde) [reply src/app/fddev/tiles/fd_benchs.c:89]: Updated, plumbed it through using `layout.quic_tile_count`.

File: src/app/fdctl/fdctl.h [hunk 3]
```diff
@@ -68,6 +68,18 @@ typedef union {
   struct {
     char name[ 13UL ];
   } flame;
+
+  struct {
+    const char * affinity;
```
Comment (mmcgee-jump) [inline src/app/fdctl/fdctl.h:73]: These should be arrays in the struct, and copy the string in there. You don't know if lifetime of those arguments.

File: src/app/fdctl/config/default.toml [hunk 4]
```diff
@@ -981,7 +981,7 @@ dynamic_port_range = "8900-9000"
         # genesis, but it might be immediately overriden at runtime if
         # the related features which increase this value are enabled.
         # The features are like update_hashes_per_tick2.
-        hashes_per_tick = 1
+        hashes_per_tick = 12500
```
Comment (mmcgee-jump) [inline src/app/fdctl/config/default.toml:984]: Let's leave this for another change. Needs more thought what the default should be (this prevents early epoch rollovers in development).

**Review by mmcgee-jump — CHANGES_REQUESTED** [review]
Nice! thanks for changes, just one minor thing then looks good

File: src/app/fdctl/fdctl.h [hunk 5]
```diff
@@ -69,6 +69,18 @@ typedef union {
   struct {
     char name[ 13UL ];
   } flame;
+
+  struct {
+    char    affinity[ AFFINITY_SZ ];
+    char    tpu_ip[ 32 ];
```
Comment (mmcgee-jump) [inline src/app/fdctl/fdctl.h:75]: These can be `uint` and do the parse in the args fn.

**Comment by mmcgee-jump** [comment]
Another nit but LGTM

**Review by mmcgee-jump — DISMISSED** [review]

**Review by mmcgee-jump — APPROVED** [review]
